src/app/courses/courses.service.ts

```diff
@@ -175,6 +176,14 @@ export class CoursesService {
   }
 
   courseAdmissionMany(courseIds, type) {
+    const addedCourseIds: any = [ ...this.userService.shelf.courseIds ];
+    console.log(addedCourseIds);
+    if (type === 'remove') {
```

Comment:
The `userService.changeShelf` function handles this.  You do not need to change `CoursesService` at all.